docker-compose-separate-containers.yml

```diff
@@ -1,23 +1,25 @@
-version: '2'
+version: '3'
 services:
   nginx:
     image: nginx
     container_name: nginx
     ports:
       - "80:80"
     volumes:
-      - /etc/nginx/conf.d
+      - nginx_conf:/etc/nginx/conf.d
 
   dockergen:
     image: jwilder/docker-gen
     command: -notify-sighup nginx -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
-    volumes_from:
-      - nginx
     volumes:
       - /var/run/docker.sock:/tmp/docker.sock:ro
       - ./nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl
+      - nginx_conf:/etc/nginx/conf.d
 
   whoami:
     image: jwilder/whoami
     environment:
       - VIRTUAL_HOST=whoami.local
+
+volumes:
+  nginx_conf: {}
```

Comment:
`nginx-conf`, because that's the convention used by Docker.

For example here: https://docs.docker.com/engine/tutorials/dockervolumes/#mount-a-shared-storage-volume-as-a-data-volume (`my-named-volume`)